src/main/java/org/commcare/formplayer/util/FormplayerDatadog.java

```diff
@@ -134,7 +124,7 @@ public void increment(String aspect, List<Tag> transientTags) {
         HashSet<String> transientKeys = new HashSet<String>();
         
         for (Tag tag : transientTags) {
-            String tagValueToSend = getTagValueToSend(tag.name, tag.value);
+            String tagValueToSend = getTagValueToSend(tag.name, tag.value, FeatureFlagChecker.isToggleEnabled("detailed_tagging"));
```

Comment:
these are the only 2 places this function is called and both pass the same FF check so you could move the FF check into the `getTagValueToSend` function itself